tests/sentry/issues/auto_source_code_config/test_process_event.py

```diff
@@ -687,10 +770,117 @@ def test_handles_dollar_sign_in_module(self) -> None:
+                ]
+            },
+            frames=[
+                {"module": "com.example.foo.Bar", "abs_path": "Bar.kt"},
+                {"module": "com.example.foo.Baz", "abs_path": "Baz.kt"},
+                {"module": "com.example.utils.Helper", "abs_path": "Helper.kt"},
+                {"module": "org.other.service.Service", "abs_path": "Service.kt"},
+            ],
+            platform=self.platform,
+            expected_new_code_mappings=[
+                {
+                    "stack_root": "com/example/foo/",
+                    "source_root": "src/com/example/foo/",
+                    "repo_name": REPO1,
+                },
+                {
+                    "stack_root": "com/example/utils/",
+                    "source_root": "src/com/example/utils/",
+                    "repo_name": REPO1,
+                },
+                {
+                    "stack_root": "org/other/service/",
+                    "source_root": "src/org/other/service/",
+                    "repo_name": REPO1,
+                },
+            ],
+            # We have three code mappings, but only two in-app stack trace rules
+            # because the "foo" code mapping coalesces with the "com.example.**" rule
+            expected_in_app_stack_trace_rules=[
+                "stack.module:com.example.** +app",
+                "stack.module:org.other.** +app",
+            ],
+            expected_num_code_mappings=3,
+        )
+
+    def test_run_without_dry_run(self) -> None:
+        repo_trees = {REPO1: ["src/com/example/foo/Bar.kt"]}
+        frames = [
+            {"module": "com.example.foo.Bar", "abs_path": "Bar.kt"},
+            {"module": "com.other.foo.Bar", "abs_path": "Bar.kt"},
+        ]
+        rule = "stack.module:com.example.**"
+        expected_in_app_rule = f"{rule} +app"
+
+        # Let's pretend we're not running as dry-run
+        with patch(f"{CODE_ROOT}.utils.PlatformConfig.is_dry_run_platform", return_value=False):
+            event = self._process_and_assert_configuration_changes(
+                repo_trees=repo_trees,
+                frames=frames,
+                platform=self.platform,
+                expected_new_code_mappings=[
+                    {
+                        "stack_root": "com/example/foo/",
+                        "source_root": "src/com/example/foo/",
+                        "repo_name": REPO1,
+                    },
+                ],
+                expected_num_code_mappings=1,
+                expected_in_app_stack_trace_rules=[expected_in_app_rule],
+            )
+            # The effects of the configuration changes will be noticed on the second event processing
+            assert event.data["metadata"]["in_app_frame_mix"] == "system-only"
+            assert len(event.data["hashes"]) == 1  # Only system hash
+            system_only_hash = event.data["hashes"][0]
+            first_enhancements_hash = event.data["grouping_config"]["enhancements"]
```

Comment:
I get why you're referring to this as a "hash", but that means something different than the "hash" on the line before. Maybe `first_enhancements_base64_string` or something?